cli/src/command/wallet.rs

```diff
@@ -128,32 +128,37 @@ pub async fn init_command(
     snapshot_path: &Path,
     parameters: InitParameters,
 ) -> Result<Wallet, Error> {
-    let password = get_password("Stronghold password", !snapshot_path.exists())?;
-    let secret_manager = SecretManager::Stronghold(
-        StrongholdSecretManager::builder()
-            .password(&password)
-            .build(snapshot_path)?,
-    );
-    let wallet = Wallet::builder()
-        .with_secret_manager(secret_manager)
-        .with_client_options(ClientOptions::new().with_node(parameters.node_url.as_str())?)
-        .with_storage_path(storage_path.to_str().expect("invalid unicode"))
-        .with_coin_type(parameters.coin_type)
-        .finish()
-        .await?;
-
+    if storage_path.exists() {
+        return Err(Error::Miscellaneous(format!(
+            "cannot initialize: {} already exists",
+            storage_path.display()
+        )));
+    }
+    if snapshot_path.exists() {
+        return Err(Error::Miscellaneous(format!(
+            "cannot initialize: {} already exists",
+            snapshot_path.display()
+        )));
+    }
```

Comment:
I'm a bit confused how this actually fixes #361, what about `The user should - again - be requested to enter a mnemonic before proceeding to the initial account creation step.` ?

Reply:
this can no longer happen. If you fully run through `init` the stronghold will contain a mnemonic. What I describe in the issue is an alternative way to go about. I explain in the PR description why I decided to it differently (basically not allowing a stronghold file with a faulty state)

Reply:
Ok, I trust you

Reply:
The "thing" that actually fixes this issue is moving the `mnemonic` creation/importing step before the secret manager build!